src/technologydata/package_data/dea_energy_storage/dea_energy_storage.py

```diff
@@ -0,0 +1,656 @@
+        The input 'est' string to be cleaned.
+
+    Returns
+    -------
+    str
+        The cleaned 'est' string.
+
+    Examples
+    --------
+    >>> clean_est_string("Lower")
+    lower
+    >>> clean_est_string("ctrl")
+    control
+
+    """
+    if est_str == "ctrl":
+        cleaned_str = "control"
+    else:
+        cleaned_str = est_str.casefold().strip()
+    return cleaned_str
+
+
+def standardize_units(series: pandas.Series) -> pandas.Series:
+    """
+    Complete missing units based on parameter names and replace incorrect units.
+
+    Parameters
+    ----------
+    series : pandas.Series
+        A series containing two elements: [par, unit]
+
+    Returns
+    -------
+    pandas.Series
+        Updated series with completed and corrected unit.
+
+    Notes
+    -----
+    The following substitutions are driven by the `pint` documentation available
+     at https://github.com/hgrecco/pint/blob/master/pint/default_en.txt:
+    - "pct.": "percent",
+    - "m2": "meter**2",
+    - "m3": "meter**3",
+
+    """
+    par, unit = series
+
+    # Mapping of parameters to their default units
+    param_unit_map = {
+        "energy storage capacity for one unit": "MWh",
+        "typical temperature difference in storage": "K",
+        "fixed o&m": "pct./year",
+        "lifetime in total number of cycles": "cycles",
+        "cycle life": "cycles",
+    }
+
+    # Mapping of incorrect units to correct units
+    unit_corrections = {
+        "pct./period": "percent",
+        "⁰C": "C",
+        "°C": "C",
+        "pct./30sec": "pct.",
+        "m2": "meter**2",
+        "m3": "meter**3",
+        "MWhoutput": "MWh",
+        "hot/cold,K": "K",
+        "pct.investement": "percent",
+        "pct.investment": "percent",
+        "tank/": "",
+        "pct.": "percent",
+    }
+
+    # Complete missing or empty units
+    if (not isinstance(unit, str)) or (unit.strip() == ""):
+        unit = param_unit_map.get(par, unit)
+
+    # Replace wrong units
+    for incorrect, correct in unit_corrections.items():
+        if incorrect == unit or incorrect in unit:
+            unit = unit.replace(incorrect, correct)
+
+    return pandas.Series([par, unit])
+
+
+def filter_parameters(
+    dataframe: pandas.DataFrame, filter_flag: bool
+) -> pandas.DataFrame:
+    """
+    Filter rows of a DataFrame by allowed technology parameters.
+
+    Parameters
+    ----------
+    dataframe : pandas.DataFrame
+        Input DataFrame containing at least a "Technology" column.
+    filter_flag : Boolean
+        If true, filter parameter `par` column
+
+    Returns
+    -------
+    pandas.DataFrame
+        The filtered DataFrame. The returned
+        DataFrame contains only rows where the `par` column is one of:
+        "technical lifetime", "fixed o&m", "specific investment", or
+        "variable o&m".
+
+    """
+    allowed_set = {
+        "technical lifetime",
+        "fixed o&m",
+        "specific investment",
+        "variable o&m",
```

Comment:
Can we also have
* `- Charge efficiency [%]`
* `- Discharge efficiency [%]`
* Storage capacities (depending on the tech it is called capacity or volume ... can be harmonised to `capacity`)

in here?

Reply:
There are some that you can all harmonise into a parameter called "capacity":
```
Energy storage capacity for one unit [kWh]
Energy storage capacity for one unit [MWh)
Energy storage capacity for one unit [MWh]
Tank volume of example [l]
Tank volume of example [m3]
```